components/core/src/clp/streaming_compression/lzma/Decompressor.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Decompressor.hpp"
+
+#include <cstddef>
+
+#include "../../ErrorCode.hpp"
+#include "../../ReaderInterface.hpp"
+#include "../../TraceableException.hpp"
+
+namespace clp::streaming_compression::lzma {
+[[nodiscard]] auto Decompressor::try_read(
```

Comment:
```suggestion
auto Decompressor::try_read(
```
You should only need `[[nodiscard]]` in the definition. Same for the rest.